background/imports/inactivity.js

```diff
@@ -0,0 +1,37 @@
+import { handleBadgeAlarm } from "../message-cache.js";
+
+const INACTIVITY_DELAY_MINS = 10;
+const INACTIVITY_ALARM = "inactivityAlarm";
+
+// This function should be called each time the user interacts with scratch.mit.edu or with the extension.
+// We consider the user inactive if it's been some minutes since this function was last called.
+export function setUserAsActive() {
+  chrome.alarms.clear(INACTIVITY_ALARM);
+  chrome.storage.session?.set({ inactivity: false });
```

Comment:
Is there a reason the key isn't simply called `active`?

Reply:
Well, at the beginning of the session we want to consider the user active, but nothing will be in storage, so "inactivity" will be undefined which is falsey.

Reply:
But if you don't want to switch all the booleans around, it could be renamed to `inactive`, sure.